provisioning_templates/cloud_init/cloud_init_default.erb

```diff
@@ -26,15 +26,19 @@ manage_etc_hosts: true
 users: {}
 runcmd:
 - |
-<%= indent(2) { snippet 'redhat_register' } %>
+<%= indent(2) { snippet 'redhat_register' } -%>
 - |
-<% if puppet_enabled %>
-<% if host_param_true?('enable-puppetlabs-repo') || host_param_true?('enable-puppetlabs-puppet6-repo') || host_param_true?('enable-puppetlabs-puppet5-repo') -%>
-<%= snippet 'puppetlabs_repo' %>
-<% end -%>
-<%= indent(2) { snippet('puppet_setup', :variables => { :full_puppet_run => true }) } %>
-<% end -%>
-
+<%=
+  if (puppet_enabled && host_param_true?('enable-puppetlabs-repo')) ||
+    host_param_true?('enable-puppetlabs-puppet6-repo') ||
+    host_param_true?('enable-puppetlabs-puppet5-repo')
+    indent(2) { snippet 'puppetlabs_repo' }
```

Comment:
still not the same logic as before. should be:
```ruby
  if puppet_enabled && (host_param_true?('enable-puppetlabs-repo') ||
    host_param_true?('enable-puppetlabs-puppet6-repo') ||
    host_param_true?('enable-puppetlabs-puppet5-repo'))

Reply:
Facepalm - I haven't looked it up.